# Origin: NetBSD pkgsrc fff4deb639a1a640476203c80f752fb77b6cb14b.
# EmberBSD AI-assisted adaptation selects official GNOME libsoup 3.8.0.
# Keep native and cross packages on the same release and GI/docs policy.

--- a/net/libsoup3/Makefile
+++ b/net/libsoup3/Makefile
@@ -1,8 +1,8 @@
 # $NetBSD: Makefile,v 1.28 2026/09/02 19:03:59 wiz Exp $
+.include "../../mk/bsd.prefs.mk"
 
-DISTNAME=	libsoup-3.6.6
+DISTNAME=	libsoup-3.8.0
 PKGNAME=	${DISTNAME:S/soup/soup3/}
-PKGREVISION=	2
 CATEGORIES=	net gnome
 MASTER_SITES=	${MASTER_SITE_GNOME:=sources/libsoup/${PKGVERSION_NOREV:R}/}
 EXTRACT_SUFX=	.tar.xz
@@ -12,7 +12,9 @@
 COMMENT=	HTTP library implementation in C (major version 3)
 LICENSE=	gnu-lgpl-v2
 
+.if ${USE_CROSS_COMPILE:Uno:tl} != "yes"
 TOOL_DEPENDS+=	glib2-introspection-[0-9]*:../../devel/glib2-introspection
+.endif
 DEPENDS+=	glib-networking-[0-9]*:../../net/glib-networking
 # not found by meson
 #TOOL_DEPENDS+=	${PYPKGPREFIX}-gi-docgen-[0-9]*:../../devel/py-gi-docgen
@@ -23,8 +25,30 @@
 USE_LANGUAGES=		c
 USE_TOOLS+=		msgfmt pkg-config
 
+.if ${USE_CROSS_COMPILE:Uno:tl} == "yes"
+# Native generators inspect the target ABI through the shared GI query path.
+DEPENDS+= glib2-introspection>=2.90.1nb1:../../devel/glib2-introspection
+EMBERBSD_GI_MESON_OPTIONS= no
+.include "../../devel/gobject-introspection/cross.mk"
+MAKE_ENV+= XDG_DATA_DIRS=${CROSS_DESTDIR:Q}${CROSS_LOCALBASE:Q}/share
+_EMBERBSD_GI_REAL_SYSROOT!= cd ${CROSS_DESTDIR:Q} && pwd -P
+BUILDLINK_TRANSFORM+= I:${_EMBERBSD_GI_REAL_SYSROOT}${CROSS_LOCALBASE}:${BUILDLINK_DIR}
+MESON_ARGS+= -Dintrospection=enabled -Dtests=false -Dtls_check=false
+TOOL_DEPENDS+= glib2-tools>=0:../../devel/glib2-tools
+MESON_BINARIES+= glib-mkenums g-ir-scanner g-ir-compiler
+MESON_BINARY.glib-mkenums= ${TOOLBASE}/bin/glib-mkenums
+MESON_BINARY.g-ir-scanner= ${TOOLBASE}/bin/g-ir-scanner
+MESON_BINARY.g-ir-compiler= ${TOOLBASE}/bin/g-ir-compiler
+MESON_CROSS_VARS+= emberbsd_gi_cross emberbsd_gi_ldd_wrapper
+MESON_CROSS.emberbsd_gi_cross= true
+MESON_CROSS.emberbsd_gi_ldd_wrapper= '${EMBERBSD_GI_FILESDIR}/elf-needed.sh'
+.endif
+
 PKGCONFIG_OVERRIDE+=		output/meson-private/libsoup-3.0.pc
 PKGCONFIG_OVERRIDE_STAGE=	pre-install
+
+# The package does not ship the optional generated API reference.
+MESON_ARGS+= -Ddocs=disabled
 
 # disable vala detection
 MESON_ARGS+=	-Dvapi=disabled
@@ -46,8 +70,10 @@
 	rm -f ${WRKSRC}/subprojects/*.wrap
 
 .include "../../archivers/brotli/buildlink3.mk"
+BUILDLINK_API_DEPENDS.zstd+=	zstd>=1.4.0
+.include "../../archivers/zstd/buildlink3.mk"
 .include "../../databases/sqlite3/buildlink3.mk"
-BUILDLINK_API_DEPENDS.glib2+=	glib2>=2.69.1
+BUILDLINK_API_DEPENDS.glib2+=	glib2>=2.70.0
 .include "../../devel/glib2/buildlink3.mk"
 BUILDLINK_API_DEPENDS.gobject-introspection+=	gobject-introspection>=0.9.5
 BUILDLINK_DEPMETHOD.gobject-introspection=	build
--- a/net/libsoup3/PLIST
+++ b/net/libsoup3/PLIST
@@ -5,6 +5,7 @@
 include/libsoup-3.0/libsoup/soup-auth-manager.h
 include/libsoup-3.0/libsoup/soup-auth.h
 include/libsoup-3.0/libsoup/soup-cache.h
+include/libsoup-3.0/libsoup/soup-compression-dictionary-request.h
 include/libsoup-3.0/libsoup/soup-content-decoder.h
 include/libsoup-3.0/libsoup/soup-content-sniffer.h
 include/libsoup-3.0/libsoup/soup-cookie-jar-db.h
@@ -44,7 +45,7 @@
 lib/girepository-1.0/Soup-3.0.typelib
 lib/libsoup-3.0.so
 lib/libsoup-3.0.so.0
-lib/libsoup-3.0.so.0.7.4
+lib/libsoup-3.0.so.0.8.0
 lib/pkgconfig/libsoup-3.0.pc
 share/gir-1.0/Soup-3.0.gir
 share/locale/ab/LC_MESSAGES/libsoup-3.0.mo
--- a/net/libsoup3/distinfo
+++ b/net/libsoup3/distinfo
@@ -1,5 +1,5 @@
 $NetBSD: distinfo,v 1.8 2026/02/15 12:43:53 adam Exp $
 
-BLAKE2s (libsoup-3.6.6.tar.xz) = 25cb4330315286c1678edf7ebe7e659141e76bea9a5156414cf5ca0dbf8ea2f5
-SHA512 (libsoup-3.6.6.tar.xz) = 4018dc6f9823fd82cde0fecbb50cd1b5dd0ff4963f92f7ea465e67faf81e71580709eec59914ddbdff317963a88e4a8024e60e44087041175bc21e04022857d2
-Size (libsoup-3.6.6.tar.xz) = 1572004 bytes
+BLAKE2s (libsoup-3.8.0.tar.xz) = 5779f7c9a4d668b11de03b88a25a94045a2cdf1aae16eed03abcc24d9e7a1fe0
+SHA512 (libsoup-3.8.0.tar.xz) = 53e62f611f635a9df28ff805176edfd0e56b1e670ac11fdf537cf7d7b29f3584bec44b8d92c0744221082a33ae2afa9a8cfd63a655b0d250e66c57ccfde67770
+Size (libsoup-3.8.0.tar.xz) = 1609676 bytes
--- a/net/libsoup3/buildlink3.mk
+++ b/net/libsoup3/buildlink3.mk
@@ -6,7 +6,7 @@
 LIBSOUP3_BUILDLINK3_MK:=
 
 BUILDLINK_API_DEPENDS.libsoup3+=	libsoup3>=3.2.2
-BUILDLINK_ABI_DEPENDS.libsoup3+=	libsoup3>=3.6.6nb2
+BUILDLINK_ABI_DEPENDS.libsoup3+=	libsoup3>=3.8.0
 BUILDLINK_PKGSRCDIR.libsoup3?=		../../net/libsoup3
 
 pkgbase := libsoup3
@@ -17,6 +17,8 @@
 .endif
 
 .include "../../archivers/brotli/buildlink3.mk"
+BUILDLINK_API_DEPENDS.zstd+=	zstd>=1.4.0
+.include "../../archivers/zstd/buildlink3.mk"
 .include "../../databases/sqlite3/buildlink3.mk"
 .include "../../devel/glib2/buildlink3.mk"
 .include "../../www/libpsl/buildlink3.mk"
